handlers/search.js

```diff
@@ -91,28 +95,36 @@ function searchPage(request, reply) {
   const filters = getFilters(query);
   const advancedSearch = query.advanced_search || (Object.keys(filters).length > 0);
 
-  trials.search(queryStr, page, perPage, filters).then((_trials) => {
-    const currentPage = page || 1;
-    const pagination = generatePaginationLinks(
-      request.url, currentPage,
-      perPage, maxPages,
-      _trials.total_count
-    );
-    const queryDescription = queryStr || 'clinical trial data';
-    const title = `Search results for ${queryDescription} in OpenTrials`;
-    const description = `OpenTrials has ${_trials.total_count} matches for
+  sources.list().then((_sources) => {
+    trials.search(queryStr, page, perPage, filters).then((_trials) => {
```

Comment:
I had one reason to not do the extraction you mentioned: we don't normally do it in handlers, so it would be unusual throughout the handlers code. Everywhere else we launch a promise (if needed), followed by a `then` block where we call `reply.view`. It's just became a standard :)